client/signup/config/flows.js

```diff
@@ -258,6 +259,41 @@ function getEntrepreneurFlowDestination() {
 	return '/setup/entrepreneur/trialAcknowledge';
 }
 
+function getGuidedOnboardingFlowDestination( dependencies ) {
+	const { onboardingSegment, siteSlug, siteId, domainItem, cartItems, refParameter } = dependencies;
+
+	if ( 'no-site' === siteSlug ) {
+		return '/home';
+	}
+	let queryParams = { siteSlug, siteId };
+
+	if ( domainItem ) {
+		queryParams = { siteId };
+	}
+
+	if ( refParameter ) {
+		queryParams.ref = refParameter;
+	}
+
+	const planSlug = cartItems?.[ 0 ]?.product_slug;
+	const planTitle = getPlan( planSlug )?.getTitle();
+
+	if (
+		onboardingSegment === 'developer-or-agency' &&
+		( planTitle === 'Creator' || planTitle === 'Entrepreneur' )
+	) {
```

Comment:
Can you please [the logic I added](https://github.com/Automattic/wp-calypso/pull/91600/files#diff-a7bd84bdbc16a9cd2414a1ac5c5d990a00617b9f9a6c678a867e1257646b9385R100-R109) here into your function since that function won't be called again.

Reply:
I updated the PR to use the plan type to match the slugs (monthly, yearly, etc) with the correct plan.